### PR TITLE
recipe for libp8_platform

### PR DESCRIPTION
**Pull request by TURX**
This is for #4638 as a dependency.

### CODE REVIEW

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,65 @@
+SUMMARY="Platform support library used by libCEC and binary add-ons for Kodi"
+DESCRIPTION="This library provides platform specific support for other libraries"
+HOMEPAGE="http://libcec.pulse-eight.com/"
+COPYRIGHT="2011-2015 Pulse-Eight Limited"
+LICENSE="GNU GPL v2"
+REVISION="1"
+
+srcGitRev="$portVersion"
+SOURCE_URI="https://github.com/Pulse-Eight/platform/archive/p8-platform-$srcGitRev.tar.gz"
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Why use "srcGitRev" here?

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+CHECKSUM_SHA256="064f8d2c358895c7e0bea9ae956f8d46f3f057772cb97f2743a11d478a0f68a0"
+SOURCE_DIR="platform-p8-platform-$srcGitRev"
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Sure it's needed?

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 3]
```diff
@@ -0,0 +1,64 @@
+SUMMARY="Platform support library used by libCEC and binary add-ons for Kodi"
+DESCRIPTION="This library provides platform specific support for other libraries"
+HOMEPAGE="http://libcec.pulse-eight.com/"
+COPYRIGHT="2011-2015 Pulse-Eight Limited"
+LICENSE="GNU GPL v2"
+REVISION="1"
+
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: No new line here

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+SOURCE_URI="https://github.com/Pulse-Eight/platform/archive/p8-platform-$portVersion.tar.gz"
+CHECKSUM_SHA256="064f8d2c358895c7e0bea9ae956f8d46f3f057772cb97f2743a11d478a0f68a0"
+SOURCE_DIR="platform-p8-platform-$portVersion"
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Sure it's needed?

Comment (TURX) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: You mean SOURCE_DIR?

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+PROVIDES="
+	libp8_platform$secondaryArchSuffix = $portVersion
+	lib:libp8_platform$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Twice? ;)

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+	cmake -DCMAKE_INSTALL_PREFIX=$prefix ..
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Checked this on gcc2hybrid?

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	make $jobArgs -j$jobs
+}
+
+INSTALL()
+{
+	cd build
+	make install -j$jobs
+
+	mkdir -p $includeDir
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Haven't checked, but isn't there an option to install the headers with cmake?

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+	mv $prefix/include/* $includeDir
+	rm -rf $prefix/include
+
+	prepareInstalledDevelLibs libp8-platform
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Only one library, so:
```diff
- 	prepareInstalledDevelLibs libp8-platform
+ 	prepareInstalledDevelLib libp8-platform
```

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 4 above)
Comment (Begasus) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Yes

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 6 above)
Comment (TURX) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: There is no x86_gcc2 in ARCHITECTURE.

Comment (Begasus) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: You have SECONDARY_ARCHITECTURES defined, for this libraries end up in $libDir instead of $prefix/lib

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 4 above)
Comment (TURX) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: There would be "Command '['git', 'commit', '-m', 'import', '-q']' returned non-zero exit status 1" if having SOURCE_DIR removed.

**Comment by TURX** [comment]
The error happens on x86_gcc2:
```
/sources/platform-p8-platform-2.1.0.1/src/util/StringUtils.cpp:33: directory `/boot/system/develop/headers/os/locale' specified in #include
```

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 7 above)
Comment (TURX) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: There is no such option.
-DCMAKE_INSTALL_INCLUDEDIR has no effect.

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 6 above)
Comment (TURX) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: -DCMAKE_INSTALL_LIBDIR=$libDir also seems to have no effect.

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 4 above)
Comment (Begasus) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Strike that, needs to stay :)

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+DESCRIPTION="This library provides platform specific support for other libraries"
+HOMEPAGE="http://libcec.pulse-eight.com/"
+COPYRIGHT="2011-2015 Pulse-Eight Limited"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/Pulse-Eight/platform/archive/p8-platform-$portVersion.tar.gz"
+CHECKSUM_SHA256="064f8d2c358895c7e0bea9ae956f8d46f3f057772cb97f2743a11d478a0f68a0"
+SOURCE_DIR="platform-p8-platform-$portVersion"
+
+ARCHITECTURES="x86_64 x86 x86_gcc2"
```
Comment (Begasus) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: Doesn't build for gcc2:
```diff
- ARCHITECTURES="x86_64 x86 x86_gcc2"
+ ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
```

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 6 above)
Comment (Begasus) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]:
```diff
-	cmake -DCMAKE_INSTALL_PREFIX=$prefix ..
+	cmake -DCMAKE_BUILD_TYPE=Release  \
+		cmake -DCMAKE_INSTALL_PREFIX=$prefix \
+		cmake -DCMAKE_INSTALL_LIBDIR=$libDir ..
```

**Comment by TURX** [comment]
I left some comments on PR 4638.

**Comment by Begasus** [comment]
LGTM

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libp8_platform$secondaryArchSuffix = $portVersion
+	lib:libp8_platform$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+PROVIDES_devel="
```
Comment (korli) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: missing newline before PROVIDES_devel

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+CHECKSUM_SHA256="064f8d2c358895c7e0bea9ae956f8d46f3f057772cb97f2743a11d478a0f68a0"
+SOURCE_DIR="platform-p8-platform-$portVersion"
+PATCHES="libp8_platform-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libp8_platform$secondaryArchSuffix = $portVersion
+	lib:libp8_platform$secondaryArchSuffix = $portVersion
```
Comment (korli) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]:
```suggestion
	lib:libp8_platform$secondaryArchSuffix = 2.1.0 compat >= 2
```

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+PROVIDES_devel="
+	libp8_platform${secondaryArchSuffix}_devel = $portVersion
+	devel:libp8_platform$secondaryArchSuffix = $portVersion
```
Comment (korli) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]:
```suggestion
	devel:libp8_platform$secondaryArchSuffix = 2.1.0 compat >= 2
```

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe (hunk 11 above)
Comment (alaviss) [reply net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: It'd be better to create a `libVersion` variable: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#using-the-libversioncompat-user-variable

File: net-libs/libp8_platform/patches/libp8_platform-2.1.0.1.patchset [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
++++ b/p8-platform.pc.in
+@@ -1,6 +1,6 @@
+ prefix=@CMAKE_INSTALL_PREFIX@
+ libdir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@
+-includedir=@CMAKE_INSTALL_PREFIX@/include
++includedir=@CMAKE_INSTALL_INCLUDEDIR@
+ DEPENDENCIES=@p8-platform_LIBRARIES@
+ 
+ Name: @p8-platform_NAME@
+diff --git a/src/posix/os-types.h b/src/posix/os-types.h
```
Comment (Begasus) [inline net-libs/libp8_platform/patches/libp8_platform-2.1.0.1.patchset]: Split this into a separate patch

File: net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+REVISION="1"
+SOURCE_URI="https://github.com/Pulse-Eight/platform/archive/p8-platform-$portVersion.tar.gz"
+CHECKSUM_SHA256="064f8d2c358895c7e0bea9ae956f8d46f3f057772cb97f2743a11d478a0f68a0"
+SOURCE_DIR="platform-p8-platform-$portVersion"
+PATCHES="libp8_platform-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersionCompat="$portVersion compat >= ${portVersion%%.*}"
```
Comment (korli) [inline net-libs/libp8_platform/libp8_platform-2.1.0.1.recipe]: The package version is a bit different from the library version:
```suggestion
libVersionCompat="2.1.0 compat >= 2"
```

**Comment by korli** [comment]
@Begasus LGTM

**Comment by Begasus** [comment]
@korli there is a p8-platform-config.cmake file, shouldn't that go to the devel package?
